apex/optimizers/fused_adam.py

```diff
@@ -150,44 +166,104 @@ def step(self, closure=None, grads=None, output_params=None, scale=None, grad_no
-                    [g_bf, p_bf, m_bf, v_bf],
-                    group['lr'],
-                    beta1,
-                    beta2,
-                    group['eps'],
-                    group['step'],
-                    self.adam_w_mode,
-                    bias_correction,
-                    group['weight_decay'],
+            # If the optimizer is capturable, then if there's a grad scaler it works
+            # on the GPU + a different multi_tensor_applier should be called
+            if self.capturable:
+                # overflow check of gradients
+                found_inf = (
+                    grad_scaler._check_inf_per_device(self)[device]
+                    if grad_scaler is not None else torch.zeros((1,), device=device)
                 )
-            if(len(g_32) > 0):
-                multi_tensor_applier(self.multi_tensor_adam,
-                                     self._dummy_overflow_buf,
-                                     [g_32, p_32, m_32, v_32],
-                                     group['lr'],
-                                     beta1,
-                                     beta2,
-                                     group['eps'],
-                                     group['step'],
-                                     self.adam_w_mode,
-                                     bias_correction,
-                                     group['weight_decay'])
+                self._dummy_overflow_buf.copy_(found_inf)
 
+                # get unscale scale factor
+                scale, inv_scale = None, None
+                if grad_scaler:
+                    scale = grad_scaler._get_scale_async()
+                    inv_scale = scale.double().reciprocal().float()
+                else:
+                    scale = torch.ones((1,), device=device)
+                    inv_scale = torch.ones((1,), device=device)
+
+                if(len(g_16) > 0):
+                    multi_tensor_applier(self.multi_tensor_adam_capturable,
+					 self._dummy_overflow_buf,
+					 [g_16, p_16, m_16, v_16],
+					 group['lr'],
+					 beta1,
+					 beta2,
+					 group['eps'],
+					 group['step'],
+					 self.adam_w_mode,
+					 bias_correction,
+					 group['weight_decay'],
+					 inv_scale)
+                if g_bf:
+                    multi_tensor_applier(
+			self.multi_tensor_adam_capturable,
+			self._dummy_overflow_buf,
+			[g_bf, p_bf, m_bf, v_bf],
+			group['lr'],
+			beta1,
+			beta2,
+			group['eps'],
+			group['step'],
+			self.adam_w_mode,
+			bias_correction,
+			group['weight_decay'],
+			inv_scale)
+
+                if(len(g_32) > 0):
+                    multi_tensor_applier(self.multi_tensor_adam_capturable,
+					 self._dummy_overflow_buf,
+					 [g_32, p_32, m_32, v_32],
+					 group['lr'],
+					 beta1,
+					 beta2,
+					 group['eps'],
+					 group['step'],
+					 self.adam_w_mode,
+					 bias_correction,
+					 group['weight_decay'],
+					 inv_scale)
+            else:
```

Comment:
the indentation doesn't look consistent. could you apply some formatter?

Reply:
Fixed